app/actions/views/channel.js

```diff
@@ -114,3 +115,30 @@ export function handlePostDraftChanged(postDraft) {
         }, getState);
     };
 }
+
+export function closeDMChannel(channel) {
+    return async(dispatch, getState) => {
+        const state = getState();
+        const userId = state.entities.users.currentId;
+
+        const preferences = [{
+            user_id: userId,
+            category: Constants.CATEGORY_DIRECT_CHANNEL_SHOW,
+            name: channel.teammate_id,
+            value: 'false'
+        }];
+
+        if (channel.isFavorite) {
+            preferences.push({
+                user_id: userId,
+                category: Constants.CATEGORY_FAVORITE_CHANNEL,
+                name: channel.id
+            });
+        }
+        savePreferences(preferences)(dispatch, getState).then(() => {
```

Comment:
This should be `deletePreferences`. You're correctly hiding the direct channel, but you're favouriting the channel here.